packages/eslint-plugin/configs/recommended.js

```diff
@@ -55,8 +55,15 @@ if ( isPackageInstalled( 'typescript' ) ) {
 				// Don't require redundant JSDoc types in TypeScript files.
 				'jsdoc/require-param-type': 'off',
 				'jsdoc/require-returns-type': 'off',
-				// Handled by TS itself.
+				// Setting "no-unused-vars" to "off" will also disable
+				// the typescript type checker. So we need to explicitly enable it,
+				// but want to ignore "unused rest siblings".
+				// @issue https://github.com/WordPress/gutenberg/issues/54305
 				'no-unused-vars': 'off',
+				'@typescript-eslint/no-unused-vars': [
+					'error',
+					{ ignoreRestSiblings: true },
```

Comment:
I flagged a valid use case in another comment. Let's allow `_`-prefixed unused vars. There appears to be configuration for vars, destructuring, args, etc. Let's start with vars, I think that should be sufficient to allow our valid code case:

```json
{
"varsIgnorePattern": "^_",
}
```

Reply:
How does that ignore pattern work. In case we have following unused vars, will only be the `_` matched or also `_foo` and being allowed?
```ts
const myFunc = ( _foo, _ ) : void => 
{
    console.log('huhu');
}

Reply:
You're absolutely right. How about a smarter regex that's aware of localization functions and experimental names? Then we can use `_` —a common convention— but disallow these known values that shouldn't be valid ignores. We should cover `__unstable` as well:

`^_(?!_$|n$|x$|nx$|_experimental|_unstable)`

Reply:
Fair enough. Since we're regularly using values that start with `_`, it's not a good ignore prefix.

I remain strongly opposed to rewriting perfectly valid and correct code in order to appease the linter. In the valid cases that are flagged here, let's just add an eslint disable comment, that's what they're for 🙂